core/src/main/java/org/kiwix/kiwixmobile/core/page/viewmodel/PageState.kt

```diff
@@ -18,11 +18,16 @@
 
 package org.kiwix.kiwixmobile.core.page.viewmodel
 
-import org.kiwix.kiwixmobile.core.page.Page
+import org.kiwix.kiwixmobile.core.page.adapter.Page
+import org.kiwix.kiwixmobile.core.page.adapter.PageRelated
 
 interface PageState {
   val pageItems: List<Page>
+  val filteredPageItems: List<PageRelated>
   val showAll: Boolean
   val currentZimId: String?
   val searchTerm: String
+  val isInSelectionState: Boolean
```

Comment:
a few of these could have default implementations to avoid repeating the implementation in the 2 states